libevmasm/RuleList.h

```diff
@@ -582,7 +582,7 @@ std::vector<SimplificationRule<Pattern>> simplificationRuleListPart7(
 	rules.push_back({
 		Builtins::BYTE(A, Builtins::SHR(B, X)),
 		[=]() -> Pattern { return A.d() < B.d() / 8 ? Word(0) : Builtins::BYTE(A.d() - B.d() / 8, X); },
-		false,
+		true,
```

Comment:
Ah, damn, I should have realized this when this was added...
This is actually an "important bugfix"!

It would probably be nicer to split this rule into two, though - one with ``A.d() < B.d() / 8`` being part of the precondition lambda, generally replacing with ``Word(0)`` and setting this flag to ``true`` and one with ``A.d() >= B.d() / 8`` as part of the precondition, always replacing with ``Builtins::BYTE(A.d() - B.d() / 8, X)`` and leaving this flag as ``false``.